storage/databases/azure_table.py

```diff
@@ -54,7 +54,9 @@ def set_error(self, partition_key, step):
     def retrieve_next_record_for_transcribing(self):
         records = self.connection.query_entities(self.table_name, num_results=1, filter="Status eq '{0}'".format(Statuses.recording_ready))
         if not records.items:
-            raise NoRecordsToProcessError()
+            records = self.connection.query_entities(self.table_name, num_results=1, filter="Status eq '{0}'".format(Statuses.transcribing))
```

Comment:
Here is the sequence of processing:
1) find one with recording_ready status
2) set the state as transcribing
3) transcribe the audio
4) save transcription
5) set state as transcribed 

if any step after 2 fails, that entry would never be reconsidered again.